File: Day161_2025/Q161_GFG.cpp
class Solution {
  public:
    int countStrings(string &s) {
        // Q. Exactly one swap
        // code here
        int arr[256] = {0};
        int count = 0;
        bool flag = false;
        for(int i = 0; i < s.size(); i++){
            count += s.size() - i - 1;
            count -= arr[s[i]];
            if(arr[s[i]] != 0)flag = true;
            arr[s[i]]++;
        }
        return count + flag;
    }
};

File: Day161_2025/Q161_GFG.py
class Solution:
    def countStrings(self, s): 
        arr = [0] * 256
        count = 0
        flag = False

        for i in range(len(s)):
            count += len(s) - i - 1
            count -= arr[ord(s[i])]
            if arr[ord(s[i])] != 0:
                flag = True
            arr[ord(s[i])] += 1

        return count + int(flag)
